jsbridge/src/main/kotlin/de/prosiebensat1digital/oasisjsbridge/JsValue.kt

```diff
@@ -307,27 +316,29 @@ internal constructor(
 
     // Create a native proxy to a JS object (blocking)
     // -> see JsValue.mapToNativeObject(check: Boolean)
-    inline fun <reified T: NativeToJsInterface> mapToNativeObjectBlocking(check: Boolean = false, context: CoroutineContext? = null): T {
+    inline fun <reified T: NativeToJsInterface> mapToNativeObjectBlocking(check: Boolean, context: CoroutineContext? = null): T {
         val jsBridge = jsBridge
                 ?: throw NativeToJsRegistrationError(T::class, customMessage = "Cannot map JS value to native object because the JS interpreter has been destroyed")
 
         return jsBridge.registerNativeToJsInterfaceBlocking(this, T::class, check, context)
     }
 
-    // Create a native proxy to a JS object (without checks)
+    // Create a native proxy to a JS object
     //
     // Notes:
-    // - the proxy object will be returned even if the JS object is invalid or does not implement
-    // all of the methods of the NativeToJsInterface
-    // - the non-suspend methods of the returned proxy object will be running in the JS thread and
-    // block the caller thread if the return value is not Unit or Deferred
-    inline fun <reified T: NativeToJsInterface> mapToNativeObject(): T {
+    // - the proxy object will be returned even if the JS object is invalid or
+    // does not implement/ all of the methods of the NativeToJsInterface
```

Comment:
That means user has to check whether method/object implements `NativeToJsInterface` correctly on his own?

Reply:
In that case, the registration won't fail if a method does not exist in JS during the registration. The user will still get an IllegalArgumentException exception when trying to call the non-existing method (see `fun testRegisterNativeToJsInterfaceSuspend()` in `JsBridgeText.kt`).